lib/DataQueue.js

```diff
@@ -0,0 +1,221 @@
+    );
+    program.addParam (
+      { value: this.library, ...QRCVDTAQParameters.LibraryName }
+    );
+    program.addParam (
+      { value: length, ...QRCVDTAQParameters.LengthOfData }
+    );
+    program.addParam (
+      { value: '', ...QRCVDTAQParameters.Data }
+    );
+    program.addParam (
+      { value: wait, ...QRCVDTAQParameters.WaitTime}
+    );
+
+    this.connection.add(program)
+    this.connection.run((error, xmlOutput) => {
+      if (error) {
+        return callback(error, null);
+      }
+      return callback(null, xmlOutput);
+    });
+  }
+
+   // alias for receive, as the API is "RCV", but jt400 uses the term "read"
+   async read(wait, callback) {
+      return this.receive(wait, callback);
+    }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Send Data Queue (QSNDDTAQ) API
+  //////////////////////////////////////////////////////////////////////////////
+
+  async send(data, callback) {
+    const program = new ProgramCall('QSNDDTAQ', { lib: 'QSYS' });
+    program.addParam({value: this.name, type: '10A'});        // Data queue name
+    program.addParam({value: this.library, type: '10A'});     // Library name
+    program.addParam({value: data.length, type: '5p0'});      // Length of data
+    program.addParam({value: data, type: `${data.length}A`}); // Data
+
+    this.connection.add(program)
+    this.connection.run((error, xmlOutput) => {
+      if (error) {
+        return callback(error, null);
+      }
+      // const result = xmlToJson(xmlOutput);
+      return callback(null, xmlOutput);
+    });
+  }
+
+  // alias for send, as the API is "SND", but jt400 uses the term "write"
+  async write(data, callback) {
+    return this.send(data, callback);
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // Clear Data Queue (QCLRDTAQ) API
+  //////////////////////////////////////////////////////////////////////////////
+
+  async clear() {
+    const program = new ProgramCall('QCLRDTAQ', { lib: 'QSYS' });
+    program.addParam(this.name,    '10A'); // Data queue name
+    program.addParam(this.library, '10A'); // Library name
+
+    this.connection.add(program)
+    this.connection.run((error, xmlOutput) => {
+      if (error) {
+        return callback(error, null);
+      }
+      // const result = xmlToJson(xmlOutput);
+      return callback(null, xmlOutput);
+    });
+  }
+
+  //////////////////////////////////////////////////////////////////////////////
+  // "Private" Functions
+  //////////////////////////////////////////////////////////////////////////////
+
+  
+  async _getDataQueueDescription(callback) {
+    const program = new ProgramCall('QMHQRDQD', { lib: 'QSYS' });
+
+    const receiverVariable = [
+      [0, '10i0'],
+      [0, '10i0'],
+      [0, '10i0']
+    ];
+
+    program.addParam ( // Receiver variable
+      { value: receiverVariable, type: 'ds', io: 'out', len: 'varLen' }
```

Comment:
receiver variable should't be an array of arrays anymore. We now use `fields` key in the `ds` object. Then fields will be an array of objects. See https://nodejs-itoolkit.readthedocs.io/en/latest/ProgramCall.html#call-the-qusrobjd-program

Reply:
Fixed in latest commit